tests/tas-elasticsearch/src/test/java/org/alfresco/elasticsearch/ElasticsearchHighlightingTests.java

```diff
@@ -0,0 +1,166 @@
+package org.alfresco.elasticsearch;
+
+import static java.util.stream.Collectors.toSet;
+
+import static org.alfresco.elasticsearch.SearchQueryService.req;
+import static org.alfresco.utility.report.log.Step.STEP;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.function.Predicate;
+
+import org.alfresco.rest.search.ResponseHighlightModel;
+import org.alfresco.rest.search.RestRequestHighlightModel;
+import org.alfresco.rest.search.SearchNodeModel;
+import org.alfresco.rest.search.SearchRequest;
+import org.alfresco.tas.AlfrescoStackInitializer;
+import org.alfresco.utility.data.DataContent;
+import org.alfresco.utility.data.DataSite;
+import org.alfresco.utility.data.DataUser;
+import org.alfresco.utility.model.FileModel;
+import org.alfresco.utility.model.FileType;
+import org.alfresco.utility.model.SiteModel;
+import org.alfresco.utility.model.TestGroup;
+import org.alfresco.utility.model.UserModel;
+import org.alfresco.utility.network.ServerHealth;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.testng.AbstractTestNGSpringContextTests;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+@ContextConfiguration (locations = "classpath:alfresco-elasticsearch-context.xml",
+        initializers = AlfrescoStackInitializer.class)
+// These are TestNG tests and the assertions are hidden in searchQueryService.
+@SuppressWarnings({"PMD.JUnit4TestShouldUseTestAnnotation", "PMD.JUnitTestsShouldIncludeAssert"})
+public class ElasticsearchHighlightingTests extends AbstractTestNGSpringContextTests
+{
+    static final Logger LOGGER = LoggerFactory.getLogger(ElasticsearchHighlightingTests.class);
+    static final String FILE_A = "fileA.txt";
+    static final String FILE_B = "fileB.txt";
+    static final String CONTENT_A = "The quick brown fox jumps over the lazy dog.";
+    static final String CONTENT_B = """
+            The lazy dog sleeps under the quick brown fox. The middle of the document is quite long:
+            Lorem ipsum dolor sit amet. In veniam tempore hic provident sunt et distinctio velit et reprehenderit
+            officiis id sapiente omnis et quisquam aliquam et porro perferendis. In sequi placeat ut quaerat voluptatem
+            ea consequuntur impedit aut eaque enim aut atque rerum.
+            The end of the document mentions the dog again!
+            """;
+
+    @Autowired
+    ServerHealth serverHealth;
```

Comment:
Are those fields used anywhere else? Shouldn't they be marked as `private`?

Reply:
I saw somewhere the idea that it's slightly harmful adding `private` modifiers to test classes because we reduce the readability without adding any useful protection (test classes should be independent anyway).

I like the idea of having less boilerplate code, but tbh I'm not convinced that I "see" the private modifier on fields anyway.

What do you think?

Reply:
I have this kind of habit to add everywhere `private` modifier, so I won't forget abut it in the future. In the other hand you are probably right - IIRC JUnit 5 has dropped `private` modifier for test methods.
I'm fine with this approach - we could agree on that and treat it as a convention - @mpichura WDYT?